lib/internal/modules/esm/default_resolve.js

```diff
@@ -45,12 +45,31 @@ if (experimentalWasmModules)
   extensionFormatMap['.wasm'] = legacyExtensionFormatMap['.wasm'] = 'wasm';
 
 function resolve(specifier, parentURL) {
+  try {
+    const parsed = new URL(specifier);
+    if (parsed.protocol === 'data:') {
+      const [ , mime ] = /^([^/]+\/[^;,]+)(;base64)?,/.exec(parsed.pathname) || [ null, null, null ];
+      const format = ({
+        'text/javascript': 'module',
+        'application/json': 'json',
+        'application/wasm': experimentalWasmModules ? 'wasm' : null
+      })[mime];
```

Comment:
```suggestion
      })[mime] || null;
```

I kind of like it better to have either `null` or `undefined` as the "no format" signifier (but not both).

Reply:
true, probably should also ensure we don't inherit from Object